File: C/0202HappyNumber.c
int squaresOfDigit(int n) {
    int sumOfSquares = 0;
    while (n > 0) {
        int digit = n % 10;
        sumOfSquares += digit * digit;
        n /= 10;
    }
    return sumOfSquares;
} 

bool isHappy(int n) {
    if (n == 1) {
        return true;
    }

    int slow = n;
    int fast = n;

    while (fast != 1) {
        slow = squaresOfDigit(slow);
        fast = squaresOfDigit(squaresOfDigit(fast));

        if ((slow == fast) && (slow != 1)) {
            return false;
        }
    }
    return true;
}

File: Python/0202HappyNumber.py
def squareOfDigits(n : int) -> int:
    sumOfSquare = 0
    while n:
        digit = n % 10
        n //= 10
        sumOfSquare += digit * digit

    return sumOfSquare

def isHappy(n : int) -> bool:
    slow = n
    fast = n

    while fast != 1:
        slow = squareOfDigits(slow)
        fast = squareOfDigits(squareOfDigits(fast))

        if (slow == fast) and (slow != 1):
            return False

    return True

n = 19
print(isHappy(n))
